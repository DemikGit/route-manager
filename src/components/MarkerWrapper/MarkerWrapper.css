.marker-wrapper__container {
  position: relative;
  display: flex;
  justify-content: space-between;
  border: 1px solid #bbbbbb;
  border-radius: 3px;
  padding: 4px;
  width: 98%;
  height: 24px;
  margin-top: 6px;
  background-color: #ffffff;
  cursor: pointer;
  box-shadow: 0px 1px 3px 0px rgba(0, 0, 0, 0.2);
}

.marker-wrapper__name {
  font-size: 14px;
  line-height: 14px;
  font-weight: 400;
  max-width: 88%;
  overflow: hidden;
  text-overflow: ellipsis;
}

.marker-wrapper__button {
  width: 35px;
  height: 35px;
  position: absolute;
  right: -7px;
  top: -6px;
  transform: rotateZ(30deg);
  overflow: hidden;
  text-overflow: ellipsis;
  cursor: pointer;
  filter: drop-shadow(0px 0px 3px rgba(0, 0, 0, 0.3));
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.marker-wrapper__button__path {
  transform: translate(10px, 10px) rotateZ(60deg);
  transform-origin: center;
}

.marker-wrapper__button__text {
  fill: #ffffff;
  font-size: 150px;
  transform-origin: center;
  transform: translate(-22px, 77px) rotateZ(-31deg);
}
