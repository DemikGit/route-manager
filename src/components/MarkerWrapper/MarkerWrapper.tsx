import * as React from 'react';
import { Component } from 'react';
import './MarkerWrapper.css'


export interface IMarkerWrapperProps {
  name: string,
  id: string,
  onPointDelete: (id: string) => void,
  onDragEnter: (event: React.SyntheticEvent<HTMLDivElement> ) => void,
  onDragStart: (event: React.SyntheticEvent<HTMLDivElement> ) => void,
}

export class MarkerWrapper extends Component<IMarkerWrapperProps, object> {
  public render() {
    const { name, onPointDelete, id, onDragEnter, onDragStart } = this.props;
    const onDelete = () => { onPointDelete(id) };
    return (
      <div
        className="marker-wrapper__container"
        onDragEnter={ onDragEnter }
        onDragStart={ onDragStart }
        draggable={ true }
      >
        <div
          title={ name }
          className="marker-wrapper__name"
        >
          { name }
        </div>
        <svg
          xmlns="http://www.w3.org/2000/svg"
          x="0" y="0" width="200px" height="200px"
          version="1.1" viewBox="0 0 200 200"
          onClick={ onDelete }
          className="marker-wrapper__button"
        >
          <linearGradient
            id="linear-gradient"
            gradientTransform="rotate(60)"
          >
            <stop offset="0%" stopColor="rgb(245, 95, 26)"/>
            <stop offset="20%" stopColor="rgb(245, 95, 26)"/>
            <stop offset="80%" stopColor="rgb(255, 100, 100)"/>
            <stop offset="100%" stopColor="rgb(255, 100, 100)"/>
          </linearGradient>
          <path
            className="marker-wrapper__button__path"
            fill="url(#linear-gradient)"
            d="M86.60254037844386 0
              L 173.20508075688772 50
              L 173.20508075688772 150
              L 86.60254037844386 200
              L 0 150
              L 0 50
              Z"
          />
          <text x="50%" y="50%" className="marker-wrapper__button__text">
            âœ–
          </text>
        </svg>
      </div>
    );
  }
}
