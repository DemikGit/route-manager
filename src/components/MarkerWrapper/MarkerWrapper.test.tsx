import { shallow, mount } from 'enzyme';
import 'jest-enzyme';
import * as React from 'react';
import { MarkerWrapper } from './MarkerWrapper';

it('renders without crashing', () => {
  const name: string = 'MarkerName';
  const id: string = 'uuid';
  const onPointDelete = jest.fn();
  const onDragEnter = jest.fn();
  const onDragStart = jest.fn();

  const wrapper = shallow(
    <MarkerWrapper
      id={ id }
      name={ name }
      onPointDelete={ onPointDelete }
      onDragEnter={ onDragEnter }
      onDragStart={ onDragStart }
    />
  );

  expect(wrapper).toMatchSnapshot();
});

it('call onPointDelete if delete button was clicked ' , () => {
  const name: string = 'MarkerName';
  const id: string = 'uuid';
  const onPointDelete = jest.fn();
  const onDragEnter = jest.fn();
  const onDragStart = jest.fn();

  const wrapper = shallow(
    <MarkerWrapper
      id={ id }
      name={ name }
      onPointDelete={ onPointDelete }
      onDragEnter={ onDragEnter }
      onDragStart={ onDragStart }
    />
  );
  const button = wrapper.find('svg');
  button.simulate('click');
  expect(onPointDelete).toBeCalled()
  expect(wrapper).toMatchSnapshot();
});

it('call onDragStart when dragging started ' , () => {
  const name: string = 'MarkerName';
  const id: string = 'uuid';
  const onPointDelete = jest.fn();
  const onDragEnter = jest.fn();
  const onDragStart = jest.fn();

  const wrapper = shallow(
    <MarkerWrapper
      id={ id }
      name={ name }
      onPointDelete={ onPointDelete }
      onDragEnter={ onDragEnter }
      onDragStart={ onDragStart }
    />
  );
  const markerContainer = wrapper.find('div').first();
  markerContainer.simulate('dragStart');
  expect(onDragStart).toBeCalled()
  expect(wrapper).toMatchSnapshot();
});


it('call onDragEnter when element was overlapped ' , () => {
  const name: string = 'MarkerName';
  const id: string = 'uuid';
  const onPointDelete = jest.fn();
  const onDragEnter = jest.fn();
  const onDragStart = jest.fn();

  const wrapper = shallow(
    <MarkerWrapper
      id={ id }
      name={ name }
      onPointDelete={ onPointDelete }
      onDragEnter={ onDragEnter }
      onDragStart={ onDragStart }
    />
  );

  const markerContainer = wrapper.find('div').first();
  markerContainer.simulate('dragEnter');
  expect(onDragEnter).toBeCalled()
  expect(wrapper).toMatchSnapshot();
});
