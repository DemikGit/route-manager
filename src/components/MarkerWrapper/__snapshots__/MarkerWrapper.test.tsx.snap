// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`call onDragEnter when element was overlapped  1`] = `
<div
  className="marker-wrapper__container"
  draggable={true}
  onDragEnter={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
    }
  }
  onDragStart={[MockFunction]}
>
  <div
    className="marker-wrapper__name"
    title="MarkerName"
  >
    MarkerName
  </div>
  <svg
    className="marker-wrapper__button"
    height="200px"
    onClick={[MockFunction]}
    version="1.1"
    viewBox="0 0 200 200"
    width="200px"
    x="0"
    xmlns="http://www.w3.org/2000/svg"
    y="0"
  >
    <linearGradient
      gradientTransform="rotate(60)"
      id="linear-gradient"
    >
      <stop
        offset="0%"
        stopColor="rgb(245, 95, 26)"
      />
      <stop
        offset="20%"
        stopColor="rgb(245, 95, 26)"
      />
      <stop
        offset="80%"
        stopColor="rgb(255, 100, 100)"
      />
      <stop
        offset="100%"
        stopColor="rgb(255, 100, 100)"
      />
    </linearGradient>
    <path
      className="marker-wrapper__button__path"
      d="M86.60254037844386 0
              L 173.20508075688772 50
              L 173.20508075688772 150
              L 86.60254037844386 200
              L 0 150
              L 0 50
              Z"
      fill="url(#linear-gradient)"
    />
    <text
      className="marker-wrapper__button__text"
      x="50%"
      y="50%"
    >
      ✖
    </text>
  </svg>
</div>
`;

exports[`call onDragStart when dragging started  1`] = `
<div
  className="marker-wrapper__container"
  draggable={true}
  onDragEnter={[MockFunction]}
  onDragStart={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
    }
  }
>
  <div
    className="marker-wrapper__name"
    title="MarkerName"
  >
    MarkerName
  </div>
  <svg
    className="marker-wrapper__button"
    height="200px"
    onClick={[MockFunction]}
    version="1.1"
    viewBox="0 0 200 200"
    width="200px"
    x="0"
    xmlns="http://www.w3.org/2000/svg"
    y="0"
  >
    <linearGradient
      gradientTransform="rotate(60)"
      id="linear-gradient"
    >
      <stop
        offset="0%"
        stopColor="rgb(245, 95, 26)"
      />
      <stop
        offset="20%"
        stopColor="rgb(245, 95, 26)"
      />
      <stop
        offset="80%"
        stopColor="rgb(255, 100, 100)"
      />
      <stop
        offset="100%"
        stopColor="rgb(255, 100, 100)"
      />
    </linearGradient>
    <path
      className="marker-wrapper__button__path"
      d="M86.60254037844386 0
              L 173.20508075688772 50
              L 173.20508075688772 150
              L 86.60254037844386 200
              L 0 150
              L 0 50
              Z"
      fill="url(#linear-gradient)"
    />
    <text
      className="marker-wrapper__button__text"
      x="50%"
      y="50%"
    >
      ✖
    </text>
  </svg>
</div>
`;

exports[`call onPointDelete if delete button was clicked  1`] = `
<div
  className="marker-wrapper__container"
  draggable={true}
  onDragEnter={[MockFunction]}
  onDragStart={[MockFunction]}
>
  <div
    className="marker-wrapper__name"
    title="MarkerName"
  >
    MarkerName
  </div>
  <svg
    className="marker-wrapper__button"
    height="200px"
    onClick={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
      }
    }
    version="1.1"
    viewBox="0 0 200 200"
    width="200px"
    x="0"
    xmlns="http://www.w3.org/2000/svg"
    y="0"
  >
    <linearGradient
      gradientTransform="rotate(60)"
      id="linear-gradient"
    >
      <stop
        offset="0%"
        stopColor="rgb(245, 95, 26)"
      />
      <stop
        offset="20%"
        stopColor="rgb(245, 95, 26)"
      />
      <stop
        offset="80%"
        stopColor="rgb(255, 100, 100)"
      />
      <stop
        offset="100%"
        stopColor="rgb(255, 100, 100)"
      />
    </linearGradient>
    <path
      className="marker-wrapper__button__path"
      d="M86.60254037844386 0
              L 173.20508075688772 50
              L 173.20508075688772 150
              L 86.60254037844386 200
              L 0 150
              L 0 50
              Z"
      fill="url(#linear-gradient)"
    />
    <text
      className="marker-wrapper__button__text"
      x="50%"
      y="50%"
    >
      ✖
    </text>
  </svg>
</div>
`;

exports[`renders without crashing 1`] = `
<div
  className="marker-wrapper__container"
  draggable={true}
  onDragEnter={[MockFunction]}
  onDragStart={[MockFunction]}
>
  <div
    className="marker-wrapper__name"
    title="MarkerName"
  >
    MarkerName
  </div>
  <svg
    className="marker-wrapper__button"
    height="200px"
    onClick={[MockFunction]}
    version="1.1"
    viewBox="0 0 200 200"
    width="200px"
    x="0"
    xmlns="http://www.w3.org/2000/svg"
    y="0"
  >
    <linearGradient
      gradientTransform="rotate(60)"
      id="linear-gradient"
    >
      <stop
        offset="0%"
        stopColor="rgb(245, 95, 26)"
      />
      <stop
        offset="20%"
        stopColor="rgb(245, 95, 26)"
      />
      <stop
        offset="80%"
        stopColor="rgb(255, 100, 100)"
      />
      <stop
        offset="100%"
        stopColor="rgb(255, 100, 100)"
      />
    </linearGradient>
    <path
      className="marker-wrapper__button__path"
      d="M86.60254037844386 0
              L 173.20508075688772 50
              L 173.20508075688772 150
              L 86.60254037844386 200
              L 0 150
              L 0 50
              Z"
      fill="url(#linear-gradient)"
    />
    <text
      className="marker-wrapper__button__text"
      x="50%"
      y="50%"
    >
      ✖
    </text>
  </svg>
</div>
`;
